earth2studio/models/dx/solarradiation_afno.py

```diff
@@ -0,0 +1,363 @@
+                )
+            )
+        )
+        ssrd_mean = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/ssrd_means.npy"
+                        )
+                    )
+                )
+            )
+        )
+        ssrd_std = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/ssrd_stds.npy"
+                        )
+                    )
+                )
+            )
+        )
+        z = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/orography.npy"
+                        )
+                    )
+                )
+            )
+        ).permute(1, 0, 2, 3)
+        z = (z - z.mean()) / z.std()
+
+        lsm = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/land_sea_mask.npy"
+                        )
+                    )
+                )
+            )
+        ).permute(1, 0, 2, 3)
+
+        sincos_latlon = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/sincos_latlon.npy"
+                        )
+                    )
+                )
+            )
+        ).permute(1, 0, 2, 3)
+        return cls(
+            model, freq, era5_mean, era5_std, ssrd_mean, ssrd_std, z, lsm, sincos_latlon
+        )
+
+    def get_sza_lonlat(
+        self, lon: np.ndarray, lat: np.ndarray
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Get longitude and latitude arrays for solar zenith angle calculation.
+
+        Parameters
+        ----------
+        lon : np.ndarray
+            Longitude array
+        lat : np.ndarray
+            Latitude array
+
+        Returns
+        -------
+        tuple[np.ndarray, np.ndarray]
+            Tuple of (longitude, latitude) arrays
+        """
+        grid = np.meshgrid(lon, lat)
+        return (grid[0].reshape(-1), grid[1].reshape(-1))
+
+    def compute_sza(self, output_coords: CoordSystem) -> torch.Tensor:
+        """Compute solar zenith angle for given coordinates.
+
+        Parameters
+        ----------
+        output_coords : CoordSystem
+            Output coordinate system
+
+        Returns
+        -------
+        torch.Tensor
+            Solar zenith angle tensor
+        """
+        lon, lat = self.get_sza_lonlat(output_coords["lon"], output_coords["lat"])
+        t = output_coords["time"] + output_coords["lead_time"]
```

Comment:
Will this create a 2D matrix?

For example if time has shape 2 and lead_time has shape 2 there are 4 unique times we need the sza for.

I think this needs something like:

```
>>> from datetime import datetime, timedelta
>>> coords = {"time": np.array([datetime(2025,1,1), datetime(2024,8,1)]), "lead_time": np.array([timedelta(hours=6), timedelta(hours=12)])}
>>> t = coords['time'][:,None] + coords['lead_time'][None,:]
>>> t= t.flatten()
>>> t
array([datetime.datetime(2025, 1, 1, 6, 0),
       datetime.datetime(2025, 1, 1, 12, 0),
       datetime.datetime(2024, 8, 1, 6, 0),
       datetime.datetime(2024, 8, 1, 12, 0)], dtype=object)
```

Reply:
Oh dang, yeah this is a bug

Reply:
The loop solution looks fine to me Oliver, not the best for perf but I think fine since these are usually getting pipelines anyway.

But can you add either a commend in the doc string of this function or preferable a check to make sure time and lead_time are of size 1 for future proofing to make sure we dont forget this limit.